### PR TITLE
Updated Readme, added Events to the Doc and did some fixes

### PR DESCRIPTION
**Pull request by Falke-Design**
I added a few new things:

- Events to the Readme
- Changed payload of the events to match together f.e. `layer`, `shape`
- Added event `pm:enable` because `pm:disable` also exists
- Added event `pm:langchange`, fired when language is changed
- Added demo event.html with all available events
- Removed `pm:raiseMarkers`, is a old relict #79 
- Fix inconsistent use of `pm:remove` on a layer 
- Added event `pm:globaldrawmodetoggled`, to have a equal event to the mode events. 
  - New functions `map.pm.Draw.`: `getActiveShape()`, `globalDrawModeEnabled()`
- Added developing steps and updated CONTRIBUTING.md

Future ideas / plans / questions:
- Add also `pm:cutmodetoggled` because it is in the edit/mode section of the controls and not clear why this uses the `pm:drawstart` or the new `pm:globaldrawmodetoggled` event
- Functionnames not equal for the `enbaled` function of the modes:
  - Edit Mode:
    - globalEditEnabled
  - Drag Mode:
    - globalDrag**Mode**Enabled
  - Removal Mode:
    - globalRemovalEnabled
- Why has `pm:markerdragstart` and `pm:markerdragend` event in Edit.Line passed indexPath and other types are not? This is inconsistent.

Fix:
#525 
#574

### CODE REVIEW

**Comment by codeofsumit** [comment]
Awesome!

Let's get to your questions:
Yes, I think `pm:cutmodetoggled` makes more sense for cutting. Let's add it but the event itself should still fire `drawstart` for backwards compatibility so we don't introduce a breaking change here.

Regarding the inconsinstent function names: Would love a normalization here. Just make sure they are backwards compatible with a comment to remove them in the next major release.

Re indexPath in `pm:markerdragstart` - because it is not available in other types than L.Edit.Line - in order to find a marker/coord in the coordinate array, a path is needed. This is not needed for Circles or Markers. IMO this can stay like this. What would be your suggestion to change here?

File: README.md [hunk 1]
```diff
@@ -378,17 +411,26 @@ map.pm.toggleGlobalRemovalMode();
...
+| :---------- | :----- | :------------------------------------------------------- | :------ |
+| pm:remove   | `e`    | Fired when a layer is removed via Removal Mode           | `layer` |
 
 The following events are available on a map instance:
 
-| Event       | Params | Description                                              |
-| :---------- | :----- | :------------------------------------------------------- |
-| pm:remove   | `e`    | Fired when a layer is removed via Removal Mode           |
-| layerremove | `e`    | Standard Leaflet event. Fired when any layer is removed. |
+| Event                         | Params | Description                                              | Output            |
```
Comment (codeofsumit) [inline README.md:429]: This should theoretically add a fourth column to the table, right? Somehow, VSCode is not showing this to me but I see the code in the README 🤔

image

Does "Output" has another effect that I'm not aware of?

File: README.md [hunk 2]
```diff
@@ -575,6 +625,46 @@ I'm adopting the Issue Management of lodash which means, feature requests get th
 You can upvote existing feature requests (or create new ones). Upvotes make me see how much a feature is requested and prioritize their implementation.
 Please see the existing [Feature Requests here](https://github.com/geoman-io/leaflet-geoman/issues?q=is%3Aissue+is%3Aclosed+label%3A%22feature+request%22+sort%3Areactions-%2B1-desc) and upvote if you want them to be implemented.
 
+### Developing
```
Comment (codeofsumit) [inline README.md:628]: 👌

File: demo/events.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
+var map = L.map('map').setView([39.74739, -105], 13);
```
Comment (codeofsumit) [inline demo/events.js:1]: I think a cypress test for all events would be great too (at least all events that can be tested)

File: README.md (hunk 1 above)
Comment (Falke-Design) [reply README.md:429]: In my editor it looks well and in [stackedit](https://stackedit.io/app#) it looks also good. I don't know what I should change 🤷‍♂️

Output is the payload

**Comment by Falke-Design** [comment]
I removed the eslint errors, sorry I didn't know about this feature.

**Pull request merged**